Neos.EventSourcedNeosAdjustments/Classes/EventSourcedRouting/NodeUriBuilder.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+namespace Neos\EventSourcedNeosAdjustments\EventSourcedRouting;
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use GuzzleHttp\Psr7\Uri;
+use Neos\EventSourcedContentRepository\Domain\Context\NodeAddress\Exception\NodeAddressCannotBeSerializedException;
+use Neos\EventSourcedContentRepository\Domain\Context\NodeAddress\NodeAddress;
+use Neos\Flow\Http\Exception as HttpException;
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Flow\Mvc\Exception\NoMatchingRouteException;
+use Neos\Flow\Mvc\Routing\Exception\MissingActionNameException;
+use Neos\Flow\Mvc\Routing\UriBuilder;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Builds URIs to nodes, taking workspace (live / shared / user) into account.
+ * Shortcut nodes will be resolved and the resulting URI will point to the shortcut target (node, asset or external URI)
```

Comment:
We also handle using "preview" URLs in backend and productive URLs on the live workspace here.

Reply:
UPDATE: "shortcut nodes" - are threy REALLY handled here? I don't see it in the code :)